lib/order.rb

```diff
@@ -7,12 +11,67 @@ def initialize(id, products)
       @products = products
     end
 
+    #imports orders in from CSV
+    def self.all
+      orders_array = []
+      CSV.read('./support/orders.csv').each do |row|
+        products_hash = {}
+        products_colon = row[1].split(";")
+        products_colon.each do |e|
+          k = e.split(":").first
+          v = e.split(":").last
+          products_hash.merge!({k => v.to_f})
+        end
+        orders_array << Order.new(row[0], products_hash)
+      end
+      return orders_array
+      # puts "This is the orders_array:"
+      # ap orders_array
+    end
+
+    def self.find(input_id)
+      ordered_stuff = {}
+      #if id > 1 && id <@orders_array.length
+        my_order = Grocery::Order.all
+        my_order.each do |order|
+          if order.id == input_id
+            ordered_stuff = order.products
+            return ordered_stuff
+          end
+        end
+        if ordered_stuff.empty?
+          raise ArgumentError.new("You did not enter a valid order number.")
+        end
+    end
+
     def total
-      # TODO: implement total
+      #Can I add in some code here to convert product value inputs (not brought in from the Csv) into floats, so that it can do math? Maybe a loop here to go through each value and convert it.
+      sum = @products.values.inject(0, :+)
+      total = sum + (sum * 0.075).round(2)
+      return total
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      if @products.keys.include?(product_name)
+        return false
+      else
+        @products[product_name] = product_price.to_f
+        return true
+      end
     end
-  end
-end
+
+    def remove_product(product_name)
+        @products.delete(product_name)
+      return @products
+    end
+  end # of class
+
+end # end of module
+
+Grocery::Order.all
```

Comment:
(duplicate comment but) Don't forget to take out statements like this before you turn in! Executing this code doesn't do anything if I'm just running the tests